macros.in

```diff
@@ -1349,5 +1349,14 @@ end
     end
 }
 
+# example autobuild macros for autotools
+%autobuild_autotools_prep() %autosetup -p1
```

Comment:
I'm not sure I understand `%prep` being part of this. It's not going to differ from one build system to another, and including it in the `%build -a` steps increases the number of spec files that won't be able to use autobuild because they need to perform some sort of pre-build shenanigans. (Running `sed` on some of the extracted files, extracting additional <code>%SOURCE<var>N</var></code> tarballs, etc...)

Reply:
Sure, the %prep part is here more for demo purposes than something we'd want to ship for real. Instead we should probably just automatically add "%autosetup -p1" %prep section to *everything* that has sources and does not have a prep. I don't understand the remark about %build -a though. If you need to do pre-build shenanigans, you do it in one %prep -a for the earliest opportunity, or in %conf -p/-a or %build -p, as appropriate.

Reply:
Actually, there was a reason %prep is a part of this: language ecosystems may have source formats outside the tarball realm, and in those cases we'd rather just have the ecosystem define %prep as they see fit, using their native tools. Ruby is an example here - rpm shouldn't have to know about .gem.

As for the rest, we could indeed just autopopulate %prep with a generic "%autosetup -p1" when Autobuild is set, but I'm not sure it's somehow better than doing it explicitly in the autobuild definition. It's just one line in the definition, and there could be other ecosystem preferences I guess. Thoughts welcome certainly.

Reply:
...and that's what the last fixup does: add autobuild defaults that will be applied if there's no build-system specific section. Only probably makes sense for %prep and %clean but the mechanism is generic so supporting the all makes no difference in code.